ui-admin/src/study/participants/enrolleeView/EnrolleeTimeline.tsx

```diff
@@ -0,0 +1,112 @@
+  return Object.keys(val).includes('eventClass')
+}
+
+const isNotification = (val: Event | Notification): val is Notification => {
+  return Object.keys(val).includes('notificationConfigId')
+}
+
+/** loads the list of notifications and events for a given enrollee and displays them in the UI */
+export default function EnrolleeTimeline({ enrollee, studyEnvContext }:
+                                           { enrollee: Enrollee, studyEnvContext: StudyEnvContextT }) {
+  const { currentEnv, study, portal, currentEnvPath } = studyEnvContext
+  const [notifications, setNotifications] = useState<Notification[]>([])
+  const [events, setEvents] = useState<Event[]>([])
+  const [tableData, setTableData] = useState<(Event | Notification)[]>([])
+  const [sorting, setSorting] = React.useState<SortingState>([{ 'id': 'createdAt', 'desc': true }])
+  const [isLoading, setIsLoading] = useState(true)
+
+  const columns: ColumnDef<Event | Notification>[] = [
+    {
+      id: 'name',
+      header: 'notification/event',
+      cell: ({ row }) => {
+        return <div>
+          {isNotification(row.original) && <NotificationConfigTypeDisplay config={row.original.notificationConfig}/>}
+          {isEvent(row.original) && _capitalize(_startCase(row.original.eventClass))}
+        </div>
+      }
+    },
+    {
+      header: 'delivery type',
+      accessorKey: 'deliveryType'
+    },
+    {
+      header: 'delivery status',
+      accessorKey: 'deliveryStatus'
+    },
+    {
+      header: 'time',
+      accessorKey: 'createdAt',
+      cell: info => instantToDefaultString(info.getValue() as number)
+    },
+    {
+      header: 'config',
+      accessorKey: 'notificationConfig',
+      cell: info => info.getValue() && <Link
+        to={notificationConfigPath(info.getValue() as NotificationConfig, currentEnvPath)}> config </Link>
+    }
+  ]
+
+  /** matches each notification to a corresponding config by id */
+  function attachConfigsToNotifications(rawNotifications: Notification[]) {
+    rawNotifications.forEach(notification => {
+      notification.notificationConfig = currentEnv.notificationConfigs
+        .find(config => config.id === notification.notificationConfigId)
+    })
+  }
+
+  useEffect(() => {
+    Api.fetchEnrolleeNotifications(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+      .then(response => {
+        attachConfigsToNotifications(response)
+        setNotifications(response)
+        Api.fetchEnrolleeEvents(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+          .then(response => {
+            setEvents(response)
+            setIsLoading(false)
+          })
+      })
+  }, [enrollee.shortcode])
+
+  useEffect(() => {
+    const newTableData: (Event | Notification)[] = []
+
+    newTableData.push(...notifications)
+    newTableData.push(...events)
+
+    setTableData(newTableData)
+  }, [events, notifications])
```

Comment:
This feels weird to me to have a useEffect that's just combining the two arrays, it's essentially storing the same data in two different state variables.  

Let's eliminate the separate state variables for setEvents and setNotifications, and do this combination and calling of `setTableData` in the hook above this one